algebra/src/fields/tweedle/tests.rs

```diff
@@ -0,0 +1,355 @@
+use crate::{
+    biginteger::BigInteger256 as BigInteger,
+    bytes::{FromBytes, ToBytes},
+    fields::{
+        tweedle::{fq::Fq, fr::Fr},
+        tests::{field_test, primefield_test},
+        Field,
+        LegendreSymbol::*,
+        PrimeField, SquareRootField,
+    },
+};
+use std::str::FromStr;
+
+#[test]
+fn test_tweedle_fr() {
+    let a: Fr = rand::random();
+    let b: Fr = rand::random();
+    field_test(a, b);
+    primefield_test::<Fr>();
+}
+
+#[test]
+fn test_tweedle_fq() {
+    let a: Fq = rand::random();
+    let b: Fq = rand::random();
+    field_test(a, b);
+    primefield_test::<Fq>();
+}
+
+#[test]
+fn test_fq_add() {
+    let f1 = Fq::from_str(
+        "18386742314266644595564329008376577163854043021652781768352795308532764650733",
```

Comment:
Where are the test vectors from?

Reply:
Do you mean values used in tests? Just a random numbers, tested with Maple before adding them to the rust implementation